docs - remote sync (#65443)
metabase/metabase: docs/installation-and-operation/remote-sync.md

> - **Development mode**: For creating and editing content. You can push changes to Git, pull changes from others, create branches, and work on feature branches. Multiple Metabase instances can be in Development mode, each working on different branches.

Reviewer: > Multiple Metabase instances can be in Development mode, each working on different branches.

this makes it seem like development mode is something you put your _metabase_ into. This is confusing because we do actually have a feature that has Metabase in development mode. I don't have a better suggestion but noting that this is confusing and I think creating a wrong impresison

- **Production mode**: For serving approved, read-only content to users. Production instances only pull changes from Git (typically from your main branch) and don't allow direct editing of synced content. You can set up auto-sync to automatically pull approved changes every five minutes.

> **Synced collections**: Special collections that are tracked in Git. When you connect a Metabase instance in Development mode, it creates a "Library" collection by default (though you can rename it). Everything inside a synced collection is versioned and synchronized with your Git repository.

Reviewer: Why plural? we only have one synced collection, right?

Author: You can have sub-collections, is the thing.

> **Synced collections must be self-contained.** Everything a dashboard or question needs must be inside the synced collections for Remote Sync to work properly. This includes questions that reference models, dashboards with questions, click behaviors linking to other content, and @ mentions in documents. Exception: questions that reference snippets can't be synced, since snippets live outside collections.

Reviewer: what about filters that pick values from other question/models?

If you need to import and export table metadata as well, check out [serialization](./serialization.md).

## Setting up Remote Sync

You'll need to be an admin to set up Remote Sync.

1. [Set up a repository to store your content](#1-set-up-a-repository-to-store-your-content)
2. [Create a personal access token for development](#2-create-a-personal-access-token-for-development)
3. [Connect your development Metabase to your repository](#3-connect-your-development-metabase-to-your-repository)
4. [Add an item to your synced collection](#4-add-an-item-to-your-synced-collection)
5. [Push your changes to your repository](#5-push-your-changes-to-your-repository)
6. [Create a personal access token for production](#6-create-a-personal-access-token-for-production)
7. [Connect your production Metabase to your repository](#7-connect-your-production-metabase-to-your-repository)

> ### 1. Set up a repository to store your content

Reviewer: I kind of liked it as H2 so i could see the steps immediately when opening the article. but I guess it also makes ToC seem very long so i'm ambivalent

Before you connect Metabase to your Git repository, create a [new GitHub repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository). Initialize the repo with a README.md.

### 2. Create a personal access token for development

Create a [fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) for your repository with these permissions:

- **Contents:** Read and write
- **Metadata:** Read-only (required)

> Copy the token immediately after generating it—you'll need to paste it into your Metabase.

Reviewer: ```suggestion
Copy the token immediately after generating it — you'll need to paste it into your Metabase.
```

:meow_evil:

> ### 3. Connect your development Metabase to your repository

Reviewer: like, if we're not talking about branches in this quickstart, then it's inconsistent with "how remote sync works" because we don't review and open PRs

4. Add your access token:

   - Paste the personal access token (PAT) you created earlier. Make sure the token has [read and write permissions](#2-create-a-personal-access-token-for-development). Metabase encrypts your token before storing it.

5. Save and test the connection:

   - Click "Save changes". Metabase will check whether it can reach your repository. If the connection fails, make sure your token has the appropriate permissions and hasn't expired. You may also have incorrectly copied and pasted the PAT, in which case you'll need to generate a new token.

### 4. Add an item to your synced collection

When you first connect in Development mode, Metabase automatically creates a special collection called "Library" that's synced with your Git repository. This is a **synced collection**—any content you add to it will be tracked in Git and can be pushed to your repository.

You can rename the Library collection if you want, and you can add sub-collections within it to organize your content.

1. Navigate to the "Library" collection in your synced collections section (look for it in the left sidebar).

> 2. Create or move content into the Library collection:

Reviewer: does this need to be two items?

Check your repository—you should see the collection with your content in it.

### 6. Create a personal access token for production

Now that you have content in your repository, you can set up your production Metabase to pull that content.

> [Create a personal access token](#2-create-a-personal-access-token-for-development) following the same steps as before, but add Contents permissions to the token that are **Read-only** (NOT write), as you only want your production Metabase reading from the repo. (Contents permissions require Metadata permissions, which GitHub will add automatically).

Reviewer: especially considering that the important part (permissions) is different than step 2

### 7. Connect your production Metabase to your repository

![Production settings](./images/production-settings.png)

> 

Reviewer: ```suggestion
In your production Metabase instance:

```

   - Click "Save changes". Metabase will verify it can reach your repository. If the connection fails, verify your token has the appropriate permissions and hasn't expired.

> 6. Pull changes or enable auto-sync (optional):

Reviewer: I think this needs some words. This is about choosing to enable autosync or not - like there isn't a radio button list where you pick "pull changes" or "enable autosync"

Maybe:

6. Choose how to update your production Metabase.
  To automatically pull changes from your main branch every five minutes,  toggle on "Auto-sync with Git". 
  If you don't toggle auto-sync, you'll need to pull changes manually from your chosen branch . You can pull changes either from this settings screen or by clicking the down arrow icon in next to the synced collection in the navigation sidebar

2. The "Megafauna Analytics" dashboard appears in production with all its questions.
3. The content is read-only for users (they can view and use it, but can't edit it).

## How synced collections work in Development mode

> Synced collections are special collections that are tracked in Git. Content in synced collections is versioned and can be pushed to or pulled from your repository.

Reviewer: ```suggestion
Synced collection is a special collections that is tracked in Git. Content in the synced collection is versioned and can be pushed to or pulled from your repository.
```

If changes don't appear after switching modes: Hard refresh your browser (Cmd/Ctrl + Shift + R).

## Pushing changes to Git

You can only push changes in a Metabase with Remote Sync set to Development mode.

### Committing your changes

> When you make changes to items in the synced collection, a yellow dot appears on your synced collection (indicating uncommitted changes). To commit your changes:

Author: ```suggestion
When you make changes to items in the synced collection, a yellow dot appears on your synced collection (indicating uncommitted changes). To commit and push your changes:
```

By default, Metabase will pull any changes (if any) from the branch you specify every five minutes. You can also manually sync as needed.

## Migrating existing content to Remote Sync

> If you already have content in your Metabase, you can gradually adopt Remote Sync. Content that lives outside synced collections remains unaffected—you can continue working with it normally while you migrate content into synced collections over time.

Reviewer: ```suggestion
If you already have content in your Metabase, you can gradually adopt Remote Sync. Content that lives outside synced collections remains unaffected — you can continue working with it normally while you migrate content into synced collections over time.
```
